src/extension/tags.c

```diff
@@ -824,16 +948,30 @@ static PHP_FUNCTION(datadog_appsec_track_user_login_failure_event)
     zend_string *user_id = NULL;
     zend_bool exists = false;
     HashTable *metadata = NULL;
-    if (zend_parse_parameters(
-            ZEND_NUM_ARGS(), "Sb|h", &user_id, &exists, &metadata) == FAILURE) {
+    zend_bool automated = false; // Don't document. Only internal usage
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "Sb|hb", &user_id, &exists,
+            &metadata, &automated) == FAILURE) {
         mlog(dd_log_warning, "Unexpected parameter combination, expected "
                              "(user_id, exists, metadata)");
         return;
     }
 
-    if (user_id == NULL || ZSTR_LEN(user_id) == 0) {
-        mlog(dd_log_warning, "Unexpected empty user id");
-        return;
+    zend_string *mode = get_DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING();
+    if (automated) {
+        if (mode && strncmp(ZSTR_VAL(mode), LSTRARG(MODE_DISABLED)) == 0) {
+            return;
+        }
+
+        if (mode && strncmp(ZSTR_VAL(mode), LSTRARG(MODE_SAFE)) == 0) {
+            if (is_user_id_sensitive(user_id)) {
+                zend_string_release(user_id);
+                user_id = ZSTR_EMPTY_ALLOC();
+            }
+
+            if (metadata != NULL && zend_array_count(metadata) > 0) {
+                zend_hash_clean(metadata);
+            }
```

Comment:
idem

Reply:
Same. In fact there should not be metedata at all when `safe` mode